File: CodeGeneratorGUI/ReactRedux/src/models/ModelMetadata.ts
import { PropMetadata } from "./PropMetadata";
export interface ModelMetadata {
  name:string;
  nameSpace:string;
  caption:string;
  props:PropMetadata[];
}

export const initModelMetadata = {
  name:'',
  nameSpace:'',
  caption:'',
  props:[],

}

File: CodeGeneratorGUI/DataAccessLayer/Models/ModelMetadata.cs
using System;

namespace DataAccessLayer.Dto
{
    public class ModelMetadata
    {
      public string Name { get; set; }
      public string NameSpace { get; set; }
      public string Caption { get; set; }
      public List<PropMetadata> Props { get; set; }

    }
}
